libpromises/evalfunction.c

```diff
@@ -6661,6 +6661,192 @@ static FnCallResult FnCallReadData(ARG_UNUSED EvalContext *ctx,
     return (FnCallResult) { FNCALL_SUCCESS, (Rval) { json, RVAL_TYPE_CONTAINER } };
 }
 
+static int ContainerComparator(JsonElement const *left_obj,
+                               JsonElement const *right_obj,
+                               void *user_data)
+{
+    assert(JsonGetContainerType(left_obj) == JSON_ELEMENT_TYPE_PRIMITIVE);
+    assert(JsonGetContainerType(right_obj) == JSON_ELEMENT_TYPE_PRIMITIVE);
```

Comment:
Now that I'm looking at this, maybe the function could get a better name like `JsonPrimitiveComparator`?